raster/src/main/scala/geotrellis/raster/BufferTile.scala

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright 2021 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster
+
+import spire.syntax.cfor._
+
+/**
+ * When combined with another BufferTile the two tiles will be aligned on (0, 0) pixel of tile center.
+ * The operation will be carried over all overlapping pixels.
+ * For instance: combining a tile padded with 5 pixels on all sides with tile padded with 3 pixels on all sides will
+ * result in buffer tile with 3 pixel padding on all sides.
+ *
+ * When combined with another BufferTile the operation will be executed over the maximum shared in
+ *
+ * TODO:
+ * - What should .map do? Map the buffer pixels or not?
+ * - toString method is friendly
+ * - mutable version makes sense
+ * - toBytes needs to encode padding size?
```

Comment:
If we're ok with the explicit work with the underlying tile:
1. map would be applied to the window only (so it is consistent with any Tile behavior, with gets and foreaches)
2. still need a friednly toString
3. yes!
4. toBytes encodes window only; we don't have any reasons to encode gridBounds to the output array 🤷, no interfaces that would support reading from it.